### PR TITLE
✨ add resource for azure functions & WAF config

### PR DESCRIPTION
**Pull request by vjeffrey**
```
azure.subscription { network  { applicationGateways { wafConfiguration }}}
```

```
azure.subscription { web { apps { functions { * } } }}
```

### CODE REVIEW

File: providers/azure/resources/azure.lr.manifest.yaml [hunk 1]
```diff
@@ -1470,6 +1483,10 @@ resources:
       properties: {}
       tags: {}
       type: {}
+      wafConfig:
```
Comment (tas50) [inline providers/azure/resources/azure.lr.manifest.yaml]: is this intention to have both?

File: providers/azure/resources/azure.lr [hunk 2]
```diff
@@ -846,6 +860,22 @@ azure.subscription.networkService.applicationGateway @defaults("id name location
   properties dict
   // Gets the attached application firewall policy
   policy() azure.subscription.networkService.applicationFirewallPolicy
+  // WAF configurations
+  wafConfiguration() []azure.subscription.networkService.wafConfig
+}
+
+// Azure Application Firewall Config
+azure.subscription.networkService.wafConfig @defaults("id name") {
```
Comment (tas50) [inline providers/azure/resources/azure.lr]: if type is just a string it might be helpful here too

File: providers/azure/resources/azure.lr [hunk 3]
```diff
@@ -70,6 +70,20 @@ azure.subscription @defaults ("name") {
   iot() azure.subscription.iotService
 }
 
+// Azure function
+private azure.subscription.webService.function @defaults("name") {
```
Comment (tas50) [inline providers/azure/resources/azure.lr]: type may be helpful here too

File: providers/azure/resources/azure.lr.manifest.yaml (hunk 1 above)
Comment (preslavgerchev) [reply providers/azure/resources/azure.lr.manifest.yaml]: nope, likely a rename that snuck its way into the manifest, removed

**Review by preslavgerchev — APPROVED** [review]

**Pull request merged**